scripts/install.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/sh
+
+set -u
+
+abort() {
+    printf "%s\n" "$@" >&2
+    exit 1
+}
+
+uname_os() {
+    local OS="$(uname -s)"
+    if [[ "${OS}" != "Darwin" && "${OS}" ]]; then
+        abort "OS ${OS} is not support, bytebase is only supported on Linux and MacOS"
+    fi
+    echo ${OS}
+}
+
+uname_arch() {
+    local ARCH=$(uname -m)
+    if [[ "${ARCH}" == "amd64" || "${ARCH}" == "x86_64" ]]; then
+        ARCH="x86_64"
+    elif [[ "${ARCH}" != "arm64" ]]; then
+        abort "Arch ${ARCH} is not support, bytebase is only supported on x86_64, amd64 and arm64"
+    fi
+    echo ${ARCH}
+}
+
+test_curl() {
+    local curl_version=$(curl --version 2>/dev/null)
+    if [ $? -ne 0 ]; then
+        abort "You must install curl before installing bytebase."
+    fi
+}
+
+test_tar() {
+    local tar_version=$(tar --version 2>/dev/null)
+    if [ $? -ne 0 ]; then
+        abort "You must install tar before installing bytebase."
+    fi
+}
+
+http_download() {
+    local local_file=$1
+    local source_url=$2
+
+    echo "Start downloading ${source_url}..."
+
+    local code=$(curl -w '%{http_code}' -L -o "${local_file}" "${source_url}")
+    if [ "$code" != "200" ]; then
+        abort "Failed to download from ${source_url}, status code: ${code}"
+    fi
+
+    echo "Completed downloading ${source_url}"
+}
+
+get_bytebase_latest_version() {
+    local version_url="https://raw.githubusercontent.com/bytebase/bytebase/main/scripts/VERSION"
```

Comment:
I was thinking to fetch the local version based on the commit.

Or do we expect the script to be used outside of the repo to always install the latest version?

If that's the case, we should comment it on the top

Reply:
> Or do we expect the script to be used outside of the repo to always install the latest version?
Yes, I think user can using `/bin/bash -c "$(curl -fsSL https://xxxxxx)"` to install Bytebase. For now, we only support using this script to install the latest version, but we can bolster it to support any release version.

Reply:
Ok, if that's the case, then maybe just create a new repo like how homebrew does https://github.com/Homebrew/install ?

Reply:
> Ok, if that's the case, then maybe just create a new repo like how homebrew does https://github.com/Homebrew/install ?

I am considering this. Will we have any advantages in creating a new repository like the homebrew did?

Reply:
Make it clear that it's for installation and later maybe uninstallation. And if it's in a separate repo, then people would not have the question of whether it's installed based on the local revision or the latest.

Reply:
I get you. I need to write a script that uninstall the bytebase installed by this script. Let's create a single repo to store these two scripts after done.